src/utils/rabbit.ts

```diff
@@ -0,0 +1,90 @@
+import { menash } from 'menashmq';
+import { logger, sleep } from '../index';
+import { MongoWatcher } from './mongoWatcher';
+import { sendFailedMsg } from '../utils/message';
+import { QueueObjectType, RabbitDataType } from '../paramTypes';
+
+/**
+ * Get rabbit health status
+ * @returns {boolean} isHealthy - true if healthy
+ */
+export function getRabbitHealthStatus(): boolean {
+  return !menash.isClosed && menash.isReady;
+}
+
+export class Rabbit {
+  rabbitData: RabbitDataType;
+  healthCheckInterval: number = 30000;
+
+  constructor(rabbitData: RabbitDataType) {
+    this.rabbitData = rabbitData;
+    if (rabbitData.healthCheckInterval) this.healthCheckInterval = rabbitData.healthCheckInterval;
+  }
+
+  /**
+   * Init rabbitmq (connection and queues)
+   */
+  async initRabbit() {
+    await this.initConnection(this.rabbitData.rabbitURI);
+    await this.initQueues(this.rabbitData.queues);
+  }
+
+  /**
+   * Healthcheck for rabbitMQ connection and reconnecting in case of a failer.
+   * @param {MongoWatcher} mongoWatcher - mongoWatcher instance
+   */
+  async connRabbitHealthChecks(mongoWatcher: MongoWatcher) {
+    while (true) {
+      if (!getRabbitHealthStatus()) {
+        // If rabbitMQ unhealthy, close current connection and try reconnect
+        await menash.close();
+        await this.initRabbit();
+        await mongoWatcher.initWatch();
+        sendFailedMsg();
+      }
+
+      await sleep(this.healthCheckInterval);
+    }
+  }
+
+  /**
+   * Creates rabbitmq connection.
+   * @param {string}  rabbituri - rabbit uri
+   */
+  async initConnection(rabbituri: string) {
+    // Initialize rabbit connection if the conn isn't ready yet
+    logger.log(`connecting to rabbitMQ on URI: ${rabbituri} ...`);
+
+    if (!menash.isReady) {
+      await menash.connect(rabbituri, { retries: this.rabbitData.rabbitRetries });
+      logger.log(`successful connection to rabbitMQ on URI: ${rabbituri}`);
+    } else {
+      logger.log(`rabbit ${rabbituri} already connected`);
+    }
+  }
+
+  /**
+   * Init rabbitmq queues and exchanges binding
+   * @param {QueueObjectType[]} queues - queues array
+   */
+  async initQueues(queues: QueueObjectType[]) {
+    await Promise.all(
+      queues.map(async (queue) => {
+        if (!(queue.name in menash.queues)) {
+          logger.log(`declare queue: ${queue.name}`);
+          const queuedeclared = await menash.declareQueue(queue.name, { durable: true });
+
+          if (queue.exchange) {
+            if (!(queue.exchange.name in menash.exchanges)) {
+              await menash.declareExchange(queue.exchange.name, queue.exchange.type);
+            }
+
+            if (menash.bindings.bindings.length < 1) {
+              await menash.bind(queue.exchange.name, queuedeclared, queue.exchange.routingKey);
+            }
+          }
+        }
+      })
+    );
+  }
```

Comment:
I think it would be better to use `menash.declareTopology`. [Menashmq README.md](https://www.npmjs.com/package/menashmq?__cf_chl_captcha_tk__=c8a6f3ed0631b00a40569e983ad3f3daea059230-1626339899-0-AVPff6Ka9z9RwKVhX1R3Wp9K5E6zFd_0Jjc3p9Oo-CrKfAgzkZUrN9DXYzUPAJ__EJDSdAGMmGPRoqwTNRMx3s_qFuQRWaokneMKU0P3miO_S4SRfWtxeB-b86Qi-jimdKgkXW15SxERDfmZ_MdM-b1Dke0n1zQriQvA0BU1ID-5jAeC4M-5p9YJh1517m2-KzPrtcb_GR9IVSNaVEyF9tXOTlL2d8Y6RN_iSwwLyAeOzCWMmDVbzPqUsdSF8gMInqVFgD3mDglAqevG-Q5I2c6nMoEcuqIChP7pOy0KzaogQuDWFZcgHBeFINDOUj-J0AX-ZqFfA4XGBu9pClWnCBrzqZV_YgxZdXcWeyqVhLJaGKknQbj4DMdnQjInCqPjW6qG8mKYl82YMcpk39b1jy7EA86RqVeeI-f82tNlDz-212oHSBGl3x2UcpbCTmK7OjeNis_rfMjm-yx4-ldq21umCy44hOPtP0E_uIcEh97mfY0DeMCfOnDYm12nTaad6baU1ki7yyI2IAme70Qqt7fn1Tm6Wzg-nBV72XptVDVM-oJ5JMAWbv-I6udkqFn_R_u5HjnOIJ9V4XfLWw2JgehCgQhcRHcdj8xp4M5PY4K1cfi7uyEadcVQNHwFNF3nm83JgfstXqIVW0UVt0MbUxQ)

Reply:
done